coderd/database/dbauthz/dbauthz.go

```diff
@@ -2854,6 +2861,13 @@ func (q *querier) GetWorkspaceAppByAgentIDAndSlug(ctx context.Context, arg datab
 	return q.db.GetWorkspaceAppByAgentIDAndSlug(ctx, arg)
 }
 
+func (q *querier) GetWorkspaceAppStatusesByAppIDs(ctx context.Context, ids []uuid.UUID) ([]database.WorkspaceAppStatus, error) {
+	if err := q.authorizeContext(ctx, policy.ActionRead, rbac.ResourceSystem); err != nil {
```

Comment:
This is a little more tricky as you've got to do the workspace_app -> workspace_agent ->  ... -> workspace dance. How about we instead fetch by (app_slugs, agent_id)? Then the authz logic essentially becomes `ws, err := FetchWorkspaceByAgentID` -> `q.authorizeContext(ctx, policy.ActionUpdate, ws)`

Reply:
That would work if it were just a single workspace being fetched, but this function is used to fetch many (e.g. list workspaces), so I believe we need to fetch all.